public/components/metrics/sidebar/sidebar.tsx

```diff
@@ -95,6 +97,11 @@ export const Sidebar = ({
   }, [selectedMetrics, selectedMetricsIds, dataSourceMDSId]);
 
   useEffect(() => {
+    // If data source is enabled but is still null prevent the invalid call
+    if (dataSourceEnabled && dataSourceMDSId === null) {
```

Comment:
if local cluster is selected, what would `dataSourceMDSId` be?

Reply:
Local cluster is just an empty string "" when selected it get sets and is no longer null